bluemira/base/constants.py

```diff
@@ -23,34 +23,153 @@
 A collection of generic physical constants, conversions, and miscellaneous constants.
 """
 
-from functools import lru_cache
 from typing import List, Union
 
 import numpy as np
 from periodictable import elements
-from pint import UnitRegistry, set_application_registry
+from pint import Context, UnitRegistry, set_application_registry
+from pint.util import UnitsContainer
 
-LOCALE = "en_GB"
-ureg = UnitRegistry(
-    fmt_locale=LOCALE, preprocessors=[lambda x: x.replace("%", " percent ")]
-)
-ureg.default_format = "~P"
+
+class BMUnitRegistry(UnitRegistry):
+    """
+    Bluemira UnitRegistry
+
+    Extra conversions:
+
+    eV <-> Kelvin
+    Pa m^3 <-> mol
+
+    Extra units:
+
+    displacements_per_atom (dpa)
+    full_power_year (fpy)
+    percent (%)
+    atomic_parts_per_million (appm)
+    USD ($)
+
+    """
+
+    def __init__(self):
+
+        super().__init__(
+            fmt_locale="en_GB",
+            preprocessors=[
+                lambda x: x.replace("%", " percent "),
+                lambda x: x.replace("$", "USD "),
+            ],
+        )
+
+        # Extra units
+        self.define("displacements_per_atom  = count = dpa")
+        self.define("full_power_year = year = fpy")
+        self.define("percent = 0.01 count")
+        self.define("atomic_parts_per_million = count * 1e-6 = appm")
+        # Other currencies need to be set up in a new context
+        self.define("USD = [currency]")
+
+        self._contexts_added = False
+
+    def _add_contexts(self, contexts=None):
+        """
+        Add new contexts to registry
+        """
+        if not self._contexts_added:
+            self.contexts = [self._energy_temperature_context(), self._flow_context()]
+
+            for c in self.contexts:
+                self.add_context(c)
+
+            self._contexts_added = True
+
+        if contexts not in (None, ()):
```

Comment:
I think you mean `if contexts`,  nope no reason. I could also be more rigorous as it should be a list of pint Contexts but then I suppose it will be harder to find errors. I'll just add typing to the function definition